src/Examine.Core/Search/QueryOptions.cs

```diff
@@ -5,10 +5,10 @@ namespace Examine.Search
     public class QueryOptions
     {
         public const int DefaultMaxResults = 500;
-        public static QueryOptions SkipTake(int skip, int? take = null) => new QueryOptions(skip, take ?? DefaultMaxResults);
-        public static QueryOptions Default { get; } = new QueryOptions(0, DefaultMaxResults);
+        public static QueryOptions SkipTake(int skip, int? take = null) => new QueryOptions(skip, take ?? DefaultMaxResults,null);
```

Comment:
This is a breaking API change and I also think this is a fairly provider specific implementation. For this to be done I think you'd have to sub class this as LuceneQueryOptions with a Timeout property and then check for this type where it might be used.